src/Microsoft.Azure.WebJobs.Host/SharedListenerAttribute.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    /// <summary>
+    /// Attribute applied to a <see cref="ITriggerBinding"/> implementation when the binding uses
+    /// a shared listener for all functions.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class)]
+    public class SharedListenerAttribute : Attribute
+    {
+        public SharedListenerAttribute(string sharedListenerId)
+        {
+            SharedListenerId = sharedListenerId;
```

Comment:
Is `null` a valid value for this ID? If so, what's the expected behavior?

Reply:
If the value is null, it won't be used - i.e. the trigger won't be treated as having a shared listener. I avoided throwing from the constructor here because in general that's not a good practice.